tests/coprocessor/test_select.rs

```diff
@@ -103,6 +104,45 @@ impl Iterator for ChunkSpliter {
     }
 }
 
+struct DAGChunkSpliter {
+    chunks: Vec<Chunk>,
+    datums: Vec<Datum>,
+    col_cnt: usize,
+}
+
+impl DAGChunkSpliter {
+    fn new(chunks: Vec<Chunk>, col_cnt: usize) -> DAGChunkSpliter {
+        DAGChunkSpliter {
+            chunks: chunks,
+            col_cnt: col_cnt,
+            datums: Vec::with_capacity(0),
+        }
+    }
+}
+
+impl Iterator for DAGChunkSpliter {
+    type Item = Vec<Datum>;
+
+    fn next(&mut self) -> Option<Vec<Datum>> {
+        loop {
+            if self.chunks.is_empty() && self.datums.is_empty() {
+                return None;
+            } else if self.datums.is_empty() {
+                let chunk = self.chunks.swap_remove(0);
+                let mut data = chunk.get_rows_data();
+                self.datums = data.decode().unwrap();
+                continue;
+            }
+            let mut cols = Vec::with_capacity(self.col_cnt);
+            for _ in 0..self.col_cnt {
+                assert_eq!(self.datums.is_empty(), false);
```

Comment:
I think you should use `assert_eq!(self.datums.len(), self.col_cnt)`  outside the loop instead. And It seems `self.dataums` is what we need in result, I'd like to set it as a temporary variable and return it directly.

Reply:
But `self.datums` aren't only one column's datums, they're datums of one chunk.
Currently i find that decode can only decode a byte slice all to datums, can't decode only part of it.

Reply:
Could we  use `assert!(self.datums.len()>=self.col_cnt)` and  `self.datum.split_off` (https://doc.rust-lang.org/std/vec/struct.Vec.html#method.split_off ) instead?